modules/library/src/main/scala/zio/elasticsearch/request/options/HasSourceFiltering.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch.request.options
+
+import zio.schema.Schema
+
+trait HasSourceFiltering[R <: HasSourceFiltering[R]] {
+  def excludes(fields: String*): R
+
+  def includes(fields: String*): R
+
+  final def includes(schema: Schema.Record[_]): R = {
```

Comment:
Can we annotate this with `@tailrec`?

Reply:
With this implementation it's not possible due to the `flatMap` call on the list, and I'm not sure it is possible with a different implementation either, because each element might call the recursive function.